crates/pico/tests/memoization.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
```

Comment:
I think it would be helpful to have some tests related to eviction. Namely, my understanding is that you are always allowed to evict derived node values from the cache and get the same results, with the only difference being that we call some of the internal functions more.

I don't think we need any special storage, we can probably just call `db.storage_mut().derived_nodes.remove(...)` or whatnot